lib/src/grid/master-checkbox.component.ts

```diff
@@ -0,0 +1,45 @@
+/* tslint:disable:component-selector */
+/* tslint:disable:no-input-rename */
+/* tslint:disable:use-host-property-decorator */
+// https://github.com/mgechev/codelyzer/issues/178#issuecomment-265154480
+import { Component, HostBinding, HostListener, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+    selector: 'th[hc-master-checkbox]',
```

Comment:
I don't love the name `hc-master-checkbox` it could possibly be `hc-grid-checkbox`?

Reply:
also, what is the value add to having a "grid-checkbox" over just using `hc-checkbox` inside the thead>tr>th? I feel removing this component simplifies things quite a bit

Reply:
Agreed about the name.

Reply:
let's talk options for `hc-grid-checkbox`. We could create a guide, we could do that work in a service instead.